src/lib/Lib/MrDocsCompilationDatabase.cpp

```diff
@@ -274,19 +277,42 @@ adjustCommandLine(
         new_cmdline.emplace_back(fmt::format("-D{}", def));
     }
 
-    // ------------------------------------------------------
-    // Add implicit include paths
-    // ------------------------------------------------------
-    // Implicit include paths are those which are automatically
-    // added by the compiler. These will not be defined in the
-    // compile command, so we add them here so that clang
-    // can also find these headers.
-    if (auto it = implicitIncludeDirectories.find(progName);
-        it != implicitIncludeDirectories.end()) {
-        for (auto const& inc : it->second)
+    if (useSystemStdLib)
+    {
+        // ------------------------------------------------------
+        // Add implicit include paths
+        // ------------------------------------------------------
+        // Implicit include paths are those which are automatically
+        // added by the compiler. These will not be defined in the
+        // compile command, so we add them here so that clang
+        // can also find these headers.
+        if (auto const it = implicitIncludeDirectories.find(progName);
```

Comment:
Besides checking the use of SystemStdLib, the input here should be empty. We don't want to calculate it if we're not using it.